mission_control/test/test_attitude_servo_behavior.py

```diff
@@ -0,0 +1,119 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+"""
+import sys
+import os
+import unittest
+import math
+import rosnode
+import rospy
+import rostest
+from mission_control.msg import ReportExecuteMissionState
+from mission_control.msg import AttitudeServo
+from pose_estimator.msg import CorrectedData
+from geometry_msgs.msg import Vector3
+from mission_interface import MissionInterface
+from mission_interface import wait_for
+
+
+class TestAttitudeServoBehavior(unittest.TestCase):
+    """
+        The test loads and execute a attitude servo mission.
+        -   Load the mission attitude_servo_mission_test.xml
+        -   Execute the mission
+        -   Check if the behavior publishes the goal
+        -   Simulate PoseEstimator data to finish the behavior
+        -   Test if the mission is SUCCESS
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_attitude_servo_behavioral')
+
+    def setUp(self):
+        self.attitude_servo_goal = AttitudeServo()
+        self.mission = MissionInterface()
+
+        # Subscribers
+        self.attitude_servo_msg = rospy.Subscriber('/mngr/attitude_servo',
+                                                   AttitudeServo, self.callback_publish_attitude_servo_goal)
+
+        self.simulated_pose_estimator_pub = rospy.Publisher('/pose/corrected_data',
+                                                            CorrectedData, queue_size=1)
+
+    def callback_publish_attitude_servo_goal(self, msg):
+        self.attitude_servo_goal = msg
+        rospy.loginfo(msg)
+
+    def test_mission_with_attitude_servo_behavioral(self):
+        self.mission.load_mission("attitude_servo_mission_test.xml")
+        self.mission.execute_mission()
+        rospy.loginfo("Ejecutando")
+
+        self.mission.read_behavior_parameters('AttitudeServoBehavior')
+        roll = self.mission.get_behavior_parameter('roll')
+        pitch = self.mission.get_behavior_parameter('pitch')
+        yaw = self.mission.get_behavior_parameter('yaw')
+        speed_knots = self.mission.get_behavior_parameter('speed_knots')
+
+        # Check if the behavior publishes the goal
+        def attitude_servo_goals_are_set():
+            return (self.attitude_servo_goal.roll == roll and
+                    self.attitude_servo_goal.pitch == pitch and
+                    self.attitude_servo_goal.yaw == yaw and
+                    self.attitude_servo_goal.speed_knots == speed_knots and
+                    self.attitude_servo_goal.ena_mask == 15)
```

Comment:
@patapufate where does this magic number come from?

Reply:
Changed to be calculated from mission parameters